alf/networks/transformer_networks.py

```diff
@@ -222,3 +227,158 @@ def forward(self, inputs, state=()):
+                the last layer. If None, a variance_scaling_initializer will be
+                used.
+            use_fc_bn (bool): whether use Batch Normalization for fc layers.
+            num_of_heads (int): number of heads for the mult-head attention
+            last_layer_size (None): nt used; for interface compatibility
+            last_activation (None): not used; for interface compatibility
+            last_kernel_initializer (None): not used; for interface compatibility
+            last_use_fc_bn (None): not used; for interface compatibility
+            name (str):
+        """
+        super().__init__(
+            input_tensor_spec,
+            input_preprocessors,
+            preprocessing_combiner=preprocessing_combiner,
+            name=name)
+
+        if kernel_initializer is None:
+            kernel_initializer = functools.partial(
+                variance_scaling_init,
+                mode='fan_in',
+                distribution='truncated_normal',
+                nonlinearity=activation)
+
+        embedding_layers = nn.ModuleList()
+        assert self._processed_input_tensor_spec.ndim == 2, (
+            "expect the "
+            "processed spec to have the shape of [entity_num, feature_dim]")
+        input_size = self._processed_input_tensor_spec.shape[-1]
+        for size in fc_layer_params:
+            embedding_layers.append(
+                layers.FC(
+                    input_size,
+                    size,
+                    activation=activation,
+                    use_bn=use_fc_bn,
+                    kernel_initializer=kernel_initializer))
+            input_size = size
+        self._embedding_layers = embedding_layers
+
+        fea_dim = input_size
+        assert fea_dim % num_of_heads == 0, "improper value for num_of_heads"
+        self._num_of_heads = num_of_heads
+        self._fea_dim_per_head = fea_dim // num_of_heads
+
+        # attention related layers
+        self._value_proj = layers.FC(
+            fea_dim,
+            fea_dim,
+            use_bias=False,
+            kernel_initializer=kernel_initializer)
+        self._key_proj = layers.FC(
+            fea_dim,
+            fea_dim,
+            use_bias=False,
+            kernel_initializer=kernel_initializer)
+        self._query_proj = layers.FC(
+            fea_dim,
+            fea_dim,
+            use_bias=False,
+            kernel_initializer=kernel_initializer)
+
+        self._simple_attention = alf.layers.SimpleAttention()
+
+    def forward(self, inputs, state=()):
+        """
+        Args:
+            inputs (Tensor):  with the shape of [B, N, d], where
+                B denotes batch size, N the number of entities, and d the
+                feature dimension
+            state (nested Tensor): states
+        Returns:
+            - Tensor: shape is [B, d'], where d' denotes the dimension of output
+                feature (fea_dim)
```

Comment:
Can be more explicit: It is the  size of the last layer in fc_layer_params (i.e. fc_layer_params[-1])

Reply:
Updated the comment according to suggestion.